lib/requests/sharing-request.js

```diff
@@ -64,19 +64,26 @@ class ConsentRequest extends AuthRequest {
    * @param res {ServerResponse}
    */
   static async get (req, res) {
-    const request = ConsentRequest.fromParams(req, res)
+    const request = SharingRequest.fromParams(req, res)
 
-    const appOrigin = request.getAppOrigin()
-    // Check if is already registered or is data browser
+    const appUrl = request.getAppUrl()
+    const appOrigin = `${appUrl.protocol}//${appUrl.host}`
```

Comment:
Could you construct this URL using setters on the object? For something this simple, it is probably OK to do it this way, but in general, there has been a lot of bugs that comes because suddenly, the protocol doesn't have a colon and stuff like that. So, I'd like to avoid string manipulation on URLs, and rather use setters on the object.

Reply:
I switched it so uses `new url.URL` instead of `url.parse` so now `appUrl.origin` is available and we don't have to construct it from strings.